app/controllers/ReportOneLoginProblemController.scala

```diff
@@ -0,0 +1,179 @@
+import util.{DateOfBirthValidator, DeskproEmailValidator, NameValidator, TaxIdentifierValidator}
+import views.html.{InternalErrorPage, ReportOneLoginProblemConfirmationPage, ReportOneLoginProblemPage}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReportOneLoginProblemFormBind {
+  private val emailValidator         = DeskproEmailValidator()
+  private val nameValidator          = NameValidator()
+  private val taxIdentifierValidator = TaxIdentifierValidator()
+  private val dateOfBirthValidator   = DateOfBirthValidator()
+
+  def form: Form[ReportOneLoginProblemForm] = Form[ReportOneLoginProblemForm](
+    mapping(
+      "name"               -> text
+        .verifying(
+          "problem_report.name.error.required",
+          name => name.nonEmpty
+        )
+        .verifying(
+          "problem_report.name.error.length",
+          name => name.length <= 70
+        )
+        .verifying(
+          "forms.name.error.invalid",
+          name => nameValidator.validate(name) || name.isEmpty
+        ),
+      "nino"               -> text
+        .verifying(
+          "one_login_problem.nino.error",
+          nino => nino.nonEmpty
+        )
+        .verifying(
+          "one_login_problem.nino.error",
+          nino => taxIdentifierValidator.validateNino(nino) || nino.isEmpty
+        ),
+      "sa-utr"             -> optional(
+        text
+          .verifying(
+            "one_login_problem.sa-utr.error",
+            saUtr => taxIdentifierValidator.validateSaUtr(saUtr)
+          )
+      ),
+      "date-of-birth"      -> mapping(
+        "day"   -> text.verifying("one_login_problem.date-of-birth.error.day", day => day.nonEmpty),
+        "month" -> text.verifying("one_login_problem.date-of-birth.error.month", month => month.nonEmpty),
+        "year"  -> text.verifying("one_login_problem.date-of-birth.error.year", year => year.nonEmpty)
+      )(DateOfBirth.apply)(d => Some(Tuple.fromProductTyped(d)))
+        .verifying("one_login_problem.date-of-birth.error.invalid", dob => dateOfBirthValidator.isValidDate(dob))
+        .verifying(
+          "one_login_problem.date-of-birth.error.future",
+          dob => dateOfBirthValidator.isNotFutureDate(dob) || !dateOfBirthValidator.isValidDate(dob)
+        ),
+      "email"              -> text
+        .verifying(
+          s"problem_report.email.error.required",
+          email => email.nonEmpty
+        )
+        .verifying(
+          s"problem_report.email.error.valid",
+          email => emailValidator.validate(email) || email.isEmpty
+        ),
+      "phone-number"       -> optional(text),
+      "address"            -> text
+        .verifying(
+          s"one_login_problem.address.error",
+          address => address.nonEmpty
+        ),
+      "contact-preference" -> of[ContactPreference],
+      "complaint"          -> optional(text),
```

Comment:
Should we include any basic validation for `complaint` and `phone-number`? Even if we do not know the values yet, maybe just a basic one to limit those two fields as max length?

Reply:
@PatrykRudnicki nice, are you OK to add that in?

Reply:
max length seems like a good shout, even if it's more than it needs to be